src/author/author.controller.ts

```diff
@@ -22,8 +23,12 @@ export class AuthorController {
   }
 
   @Get()
-  findAll(@Query() searchDto: SearchDto) {
-    return this.authorService.findAll(searchDto.search);
+  findAll(@Query('search') search?: string) {
+    if (search) {
```

Comment:
you dont need if and else you can write simillar : return this.authorService.findAllSearch(search);